slack-notify.sh

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+if [[ -z "${SLACK_TOKEN-}" ]]; then
+    read -r SLACK_TOKEN
+fi
+SLACK_HOOK_URL="https://hooks.slack.com/services/${SLACK_TOKEN}"
+
+: "${GITHUB_SERVER_URL:=https://github.com}"
+: "${GITHUB_API_URL:=https://api.github.com}"
+: "${GITHUB_SHA:=$(git rev-parse HEAD)}"
+: "${PICTURE_BASE_URL:=https://github.com/foxgoat/howl/raw/master}"
+
+if [[ -z "${GITHUB_REPOSITORY-}" ]]; then
+    REPO_URL=$(git remote get-url origin)
+    GITHUB_REPOSITORY="${REPO_URL#${GITHUB_SERVER_URL}/}"
+else
+    REPO_URL="${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}"
+fi
+REPO=${GITHUB_REPOSITORY#*/}
+
+: "${BUILD_URL:=${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}/actions${GITHUB_RUN_ID:+/runs/${GITHUB_RUN_ID}}}"
+: "${GITHUB_REF:=$(git symbolic-ref HEAD)}"
+BRANCH=${GITHUB_REF#refs/heads/}
+BRANCH_URL="${REPO_URL}/tree/${BRANCH}"
+
+# prop extracts property from JSON object.
+# prop('{"name": "Tricia", "age": 42}', age)
+# 42
+prop() {
+    jq -r ".$2" <<<"$1"
+}
+
+CURL_HEADERS=(-H "Accept: application/vnd.github.groot-preview+json")
+if [[ -n "${GITHUB_TOKEN-}" ]]; then
+    CURL_HEADERS+=(-H "Authorization: token ${GITHUB_TOKEN}")
+fi
+
+COMMIT_URL="${GITHUB_API_URL}/repos/${GITHUB_REPOSITORY}/commits/${GITHUB_SHA}"
+pr=$(curl -fsSL "${CURL_HEADERS[@]}" "${COMMIT_URL}/pulls" |
+    jq '.[0] | {url: .html_url, number, title: .title, author: .user.login}' || echo '{}')
```

Comment:
can `title: .title` just be `title` like it is for `number`?

Reply:
yes it was left over from changing it from message. thanks!